goas/zone/common/src/lib.rs

```diff
@@ -79,22 +93,30 @@ impl Withdraw {
     }
 }
 
-#[derive(Debug, Clone, Copy, Serialize, Deserialize)]
-pub enum Input {
-    Withdraw(Withdraw),
+/// A deposit of funds into the zone
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+pub struct Deposit {
+    /// The note that is used to deposit funds into the zone
+    pub deposit: InputWitness,
+
+    // This zone state note
+    pub zone_note_in: InputWitness,
+    pub zone_note_out: OutputWitness,
+
+    // The zone funds note
+    pub zone_funds_in: InputWitness,
```

Comment:
weren't we going to drop the zone_funds_in from the deposit transaction?

ie. deposits always produce new zone fund notes?

Reply:
The first version had it optional, and then I figured it out it would be better to force merging notes if we can?

Reply:
I'd drop it if it's not necessary for correctness, it just adds more constraints that we need to check in the zone's state death constraint

Reply:
discussed offline, both approaches will work, this will prevent a proliferation of zone fund notes which is nice so we'll go with this for now.